pkg/domain/events.go

```diff
@@ -0,0 +1,106 @@
+package domain
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/util"
+)
+
+type Event interface {
+	Name() string
+	Message() string
+}
+
+type BlueprintSpecInvalidEvent struct {
+	ValidationError error
+}
+
+func (b BlueprintSpecInvalidEvent) Name() string {
+	return "BlueprintSpecInvalid"
+}
+
+func (b BlueprintSpecInvalidEvent) Message() string {
+	return b.ValidationError.Error()
+}
+
+type BlueprintSpecStaticallyValidatedEvent struct{}
+
+func (b BlueprintSpecStaticallyValidatedEvent) Name() string {
+	return "BlueprintSpecStaticallyValidated"
+}
+
+func (b BlueprintSpecStaticallyValidatedEvent) Message() string {
+	return ""
+}
+
+type BlueprintSpecValidatedEvent struct{}
+
+func (b BlueprintSpecValidatedEvent) Name() string {
+	return "BlueprintSpecValidated"
+}
+
+func (b BlueprintSpecValidatedEvent) Message() string {
+	return ""
+}
+
+type DogusHealthyEvent struct{}
+
+func (d DogusHealthyEvent) Name() string {
+	return "DogusHealthy"
+}
+
+func (d DogusHealthyEvent) Message() string {
+	return ""
+}
+
+type IgnoreDoguHealthEvent struct{}
+
+func (i IgnoreDoguHealthEvent) Name() string {
+	return "IgnoreDoguHealth"
+}
+
+func (i IgnoreDoguHealthEvent) Message() string {
+	return "ignore dogu health flag is set; ignoring dogu health"
+}
+
+type DogusUnhealthyEvent struct {
+	HealthResult ecosystem.DoguHealthResult
+}
+
+func (d DogusUnhealthyEvent) Name() string {
+	return "DogusUnhealthy"
+}
+
+func (d DogusUnhealthyEvent) Message() string {
+	unhealthyDogus := util.Map(d.HealthResult.UnhealthyDogus, ecosystem.UnhealthyDogu.String)
+	slices.Sort(unhealthyDogus)
+	return fmt.Sprintf("%d dogus are unhealthy: %s", len(unhealthyDogus),
+		strings.Join(unhealthyDogus, ", "))
+}
+
+type EffectiveBlueprintCalculatedEvent struct{}
```

Comment:
Events should contain the results of the command (if there are any) because in DDD domain events are made to be consumed by others eventually. 
I know, we don't do that at the moment and don't use the result in the event message but i think we should provide the results so that we don't need to change the domain if anybody wants to consume it later.